erpnext/accounts/doctype/asset/asset.py

```diff
@@ -80,7 +80,7 @@ def validate_asset_values(self):
 			if cint(self.number_of_depreciations_booked) > cint(self.total_number_of_depreciations):
 				frappe.throw(_("Number of Depreciations Booked cannot be greater than Total Number of Depreciations"))
 						
-		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(nowdate()):
+		if self.next_depreciation_date and getdate(self.next_depreciation_date) < getdate(self.purchase_date):
```

Comment:
Thanks for the insight.
Yes you are right, and i guess there should be some indication in asset after saving that their past depreciation entry are not booked.

So i will add "Warning!" text to it and frappe.throw for self.purchase_date as a seperate validation.